library/README.md

```diff
@@ -1,6 +1,66 @@
 # ipfs-client
 
-## TODO
+## Usage
 
-Need to fill out this README to explain how to use ipfs-client in other contexts.
+For a rough example of how this is done, see [this command-line tester](https://github.com/John-LittleBearLabs/ipfs_client_clitester)
 
+### Instantiate ipfs::Client
+
+Using code should center on the `ipfs::Client` class found in `include/ipfs_client/client.h`
+There are a number of customization points for that class, particularly those things which might vary depending on 
+which dependencies you'd like to use and/or which behaviors you'd like. 
+
+Some concrete classes are provided with the library, each turned off if the corresponding dependency does not appear to 
+be available. If you're using the Conan package, and not on Windows, all those libraries should be already linked in for you. 
+In that case, you may use `ipfs::start_default` found in `include/ipfs_client/opinionated_context.h` , which is suitable for some non-critical uses.
+
+If any of these choices don't make sense for your project, you may create a concrete class for whichever interface you need to alter, and call `with()` on the Client object, passing in an instance of your class.
+If you don't call `with()`, usually you get an implementation that simply doesn't do anything (null object), or does something very simplistic.
+
+To see the sorts of extension points available, look at the various `with()` methods in `ipfs::Client`. For example:
+
+| Interface                         | Function                                                             | Provided example implementation                         | Design choice in example                                                                                                |
+|-----------------------------------|----------------------------------------------------------------------|---------------------------------------------------------|-------------------------------------------------------------------------------------------------------------------------|
+| ipfs::ctx::HttpApi                | Make HTTP(s) requests                                                | ipfs::ctx::BoostBeastHttp                               | Uses [Boost.Beast](https://www.boost.org/doc/libs/1_84_0/libs/beast/doc/html/index.html)                                |
+| ipfs::ctx::DnsTxtLookup           | DNS TXT requests (not name resolution)                               | ipfs::ctx::AresDnsTxtLookup                             | Uses [c-ares](https://c-ares.org/)                                                                                      |
```

Comment:
Bit confused what you mean by "(not name resolution)".
Is this to indicate "not IPNS record resolution" or "(not DNS A record)"?

Reply:
Not resolving a DNS name to an IP endpoint. i.e. not like `gethostbyname` . In this case I'm really saying this won't be used to resolve hostnames of the gateways. The HttpApi does that.